Reformat main class

BengRhui opened the pull request:
@cmchan04 I've reorganized the main class (even though it still looks a little bit messy but I guess I'll just leave it as it is...), maybe you can try n check if there's anything to modify?

If got anything tell me lah, oppa chun ming, I'll modify it (^_^)

BengRhui committed:
Check if list is null

BengRhui committed:
Add method to separate list based on channel

BengRhui committed:
Update variable name for split to channel

BengRhui committed:
Added a parenthesis for export JSON

BengRhui committed:
Change variable to pointer address

BengRhui committed:
Update sorting criteria for quick sort

BengRhui committed:
Update main program

BengRhui committed:
Update JSON methods

To reduce memory, data is loaded and written one by one instead of writing everything at once.

BengRhui committed:
Rewrite quick sort

BengRhui committed:
Add print statements in main

BengRhui commented:
@cmchan04 I've updated the quick sort and JSON, hopefully everything is runnable by now. Do try it out on your side ya oppa chun ming >3<

BengRhui committed:
Update linked list destructor

Removed the const keyword

BengRhui committed:
Delete objects

Forgot to delete some of the objects here

cmchan04 reviewed (changes requested):
@BengRhui  I remember nicholas said we are not allowed to use ```<list>``` library, if the <list> in main is not used, I think better to replace with ```<iostream>``` should be better, just avoid if he misthink we use it as data structure. I will run the code test in a while

cmchan04 commented:
There's still bad::alloc problem when it comes to the copy Array for 4 sorting. I think we can just run one by one

BengRhui committed:
Remove list and change to iostream

BengRhui commented:
> There's still bad::alloc problem when it comes to the copy Array for 4 sorting. I think we can just run one by one

I think this one we can just comment out the export JSON commands when we demo later. But anyways I've changed the imports from `list` to `iostream`, not sure why list appears but I did not use it at all haha, maybe you can try check again @cmchan04

BengRhui committed:
Update binary search

Used the wrong search for binary search haih

BengRhui committed:
Update main.cpp

cmchan04 reviewed (commented):
LGTM

BengRhui committed:
Update json generation

This properly indents the elements in JSON with 4 spacing

BengRhui committed:
Reduce memory consumption for cleaning ID

Using substring helps to reduce the memory generated, according to ChatGPT and Claude

cmchan04 commented:
LGTM. If there is nothing else, I will merge it to main tmr 12pm
